common/analysis/lint_rule_status.h

```diff
@@ -63,17 +64,21 @@ struct LintViolation {
   // The context (list of ancestors) of the offending token.
   // For non-syntax-tree analyses, leave this blank.
   const SyntaxTreeContext context;
+
+  bool operator<(const LintViolation& r) const {
+    return (token.text.data() < r.token.text.data());
```

Comment:
// compares addresses of violations, which correspond to substring locations

Reply:
sure, added the comment